### DISCUSSION TITLE
AZURE SQL Database Dynamic Ip address

### DISCUSSION THREAD

**Discussion by PatternsTechGit** [community/community#19549] [Projects and Issues] [answered]
I have an Azure SQL Database and my clients want to access it. They belong to a place where they are assigned new IP every time whenever they turn on PC/Internet. I have tried to assign a range to access it but they have a huge range gap, for eg `35.111.111.0` to `199.255.255.255`, and If I assign it full range access it will be highly insecure. Please guide me if there are some permissions or solutions I can use for this scenario.

**Comment by rajaastahir** [comment, marked as answer]
You can set IP address range from which communications are allowed into SQL Database in the Azure SQL Database firewall but for stable IP addresses, this strategy is acceptable.

A firewall security feature called virtual network rules determines whether communications from certain subnets in virtual networks are accepted by the server for your databases and elastic pools in Azure SQL Database.

For dynamic IP addresses to establish and to manage connection from a specific subnet the Virtual Network Rule can be easier substitute.
